src/bd/amazed/pdfscissors/model/TaskPdfOpen.java

```diff
@@ -54,27 +50,19 @@ public TaskPdfOpen(File file, int groupType, boolean shouldCreateStackView, Comp
 
 			if (shouldCreateStackView && groupType != PageGroup.GROUP_TYPE_INDIVIDUAL) {
 				setProgress(0);
-				PropertyChangeListener propertyChangeListener = new PropertyChangeListener() {
-					@Override
-					public void propertyChange(PropertyChangeEvent evt) {
-						firePropertyChange(evt.getPropertyName(), evt.getOldValue(), evt.getNewValue());
-					}
-				};
-
-				for (int i = 0; i < pageGroups.size(); i++) {
-					PageGroup pageGroup = pageGroups.elementAt(i);
-					BufferedImage image = cropper.getImage(propertyChangeListener, pageGroup);
-
-					if (image == null) {
-						debug("Ups.. null image for " + pdfFile.getNormalizedFile());
-					} else {
-						debug("PDF loaded " + pageGroup + " from " + pdfFile.getNormalizedFile());
-					}
-					pageGroup.setStackImage(image);
-
-				}
+				TaskCreateThumbnails thumbnailTask = launchThumbnailTask(pdfFile, pageGroups);
+				
+				//make this thread wait, until the thumbnails-thread has finished:
+				// (images are previews for stacked-pages, i.e. required)
+				thumbnailTask.get();
+				
 				setProgress(100);
 			}
+			else {
+				//start thumbnails-thread without waiting for it to finish
+				// (images are only previews for single pages, i.e. not required)
+				launchThumbnailTask(pdfFile, pageGroups);

```

Comment:
if I remember correctly (-> it's been a while since I wrote that code), this should not be a problem:

the thumbnails are now created in a separate thread (i.e. does not necessarily block the GUI thread).
When opening a file there are 2 cases:
(1) with stacked-view: then the opening dialog will wait until thumbnails have been created (ie. wait for the thumbnail-thread to finish) and then continues
(2) non-stacked-view: then it opens immediately and not-yet-created thumbnails will get a placeholder until the corresponding thumbnail was created in the thumbnails-thread

Thumbnails are now cached (-> for the main view) and not created each time upon rendering -- that is the reason why (2) also uses the thumbnail-thread, but does not wait until it has finished before showing the main view